src/Execution/Utils/Subscription.php

```diff
@@ -25,7 +25,9 @@ public static function broadcast(string $subscriptionField, $root, ?bool $should
     {
         // Ensure we have a schema and registered subscription fields
         // in the event we are calling this method in code.
-        app(GraphQL::class)->prepSchema();
+        /** @var \Nuwave\Lighthouse\GraphQL $graphQL */
+        $graphQL = app(GraphQL::class);
```

Comment:
Why was this split into a variable instead of inline it?

Reply:
To enable autocompletion via the annotation.

Reply:
Interesting fact: tools like https://github.com/barryvdh/laravel-ide-helper generate a meta file which tells (at least PhpStorm) that `app()` (and its over variants) are factory function and that they return what is passed in.

That said: I've no idea if/how this would work within a library project and I'm certainly not challenging this change!

Reply:
@mfn Yes if you just want autocompletion for phpstorm you can add in a `.phpstorm.meta.php`, and in that do a simple
```
namespace PHPSTORM_META {
   override(\app(0), map([
      ...
   ])
}
```
and that also works in packages.

However we are actually using a helper method here, which could be nice if we could remove that all together instead.
Could we just supply the classes it needs in the constructor or supply the container in the constructor of the class?

Reply:
https://github.com/nuwave/lighthouse/issues/654

Reply:
> Could we just supply the classes it needs in the constructor or supply the container in the constructor of the class?

Well in this case this is a static method so I guess constructor injection isn't possible 😅

Reply:
Oh did not realise that. 

Yeah I guess it would be weird that the method took them in as arguments.